src/oscar/apps/order/receivers.py

```diff
@@ -0,0 +1,9 @@
+# coding=utf-8
+from django.dispatch import receiver
+
+from .signals import order_status_changed
+
+
+@receiver(order_status_changed)
+def receive_order_status_change(sender, order, old_status, new_status, **kwargs):
```

Comment:
I would prefer to do this in an `Order.create_order_status_change()` method that is called directly from `Order.set_status()`. That makes is much easier to override the behaviour instead of having to unhook signals etc.  @craigloftus @sasha0 do you have any thoughts?

Reply:
Yeah, I think we're increasingly learning that signals are an anti-pattern

Reply:
Okay. Just do be clear, it is an instance method of AbstractOrder and not a classmethod?

Reply:
Yes please. This allows access to the order object for anyone that wants to modify how status changes are handled.